src/main/java/org/testng/TestRunner.java

```diff
@@ -961,7 +970,18 @@ public ISuite getSuite() {
 
   @Override
   public ITestNGMethod[] getAllTestMethods() {
-    return m_allTestMethods;
+    if (m_allTestMethods != null) {
+      //This is true only when we are running JUnit mode
+      return m_allTestMethods;
```

Comment:
Has it is only used for JUnit, rename it `m_allJunitTestMethods`

Reply:
fixed